runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java

```diff
@@ -395,77 +403,64 @@ public void stop(boolean shutdown) {
      * @param message The message to send.
      * @param <T>     The type of completable to return.
      * @return A completable future which will be fulfilled by the reply,
-     *     or a timeout in the case there is no response.
+     * or a timeout in the case there is no response.
      */
-    public <T> CompletableFuture<T> sendMessageAndGetCompletable(ChannelHandlerContext ctx,
-        @NonNull CorfuMsg message) {
-
-        // Check the connection future. If connected, continue with sending the message.
-        // If timed out, return a exceptionally completed with the timeout.
-        try {
-            connectionFuture
-                .get(parameters.getConnectionTimeout().toMillis(), TimeUnit.MILLISECONDS);
-        } catch (InterruptedException e) {
-            throw new UnrecoverableCorfuInterruptedError(e);
-        } catch (TimeoutException te) {
-            CompletableFuture<T> f = new CompletableFuture<>();
-            f.completeExceptionally(te);
-            return f;
-        } catch (ExecutionException ee) {
-            CompletableFuture<T> f = new CompletableFuture<>();
-            f.completeExceptionally(ee.getCause());
-            return f;
-        }
-
-        // Set up the timer and context to measure request
-        final Timer roundTripMsgTimer = CorfuRuntime.getDefaultMetrics()
-                .timer(timerNameCache.get(message.getMsgType()));
-
-        final Timer.Context roundTripMsgContext = MetricsUtils
-                .getConditionalContext(roundTripMsgTimer);
+    public <T> CompletableFuture<T> sendMessageAndGetCompletable(
+            ChannelHandlerContext ctx, @NonNull CorfuMsg message) {
```

Comment:
I think a refactor is needed, but don't think this is "it".

The complexity of the method hasn't been reduced. For example, a lot of superfluous code related to "timing" a request is still there. Why is a `cfBenchmarked` even needed? there are better ways of timing the round trip.

Reply:
cfBenchmarked is used to measure how much time takes a request. 
You can implement that `better way` it's up to you. 
I don't see how this change is not a "refactoring that is needed", this change deletes awkward  Cfutil with a bunch of try-catch blocks and makes the method more readable and understandable and easier to change.